presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoPreparedStatement.java

```diff
@@ -849,4 +862,65 @@ private static String typedNull(String prestoType)
     {
         return format("CAST(NULL AS %s)", prestoType);
     }
+
+    private static List<ColumnInfo> getDescribeOutputColumnInfoList(ResultSet resultSet)
+            throws SQLException
+    {
+        ImmutableList.Builder<ColumnInfo> list = ImmutableList.builder();
+        while (resultSet.next()) {
+            String columnName = resultSet.getString("Column Name");
+            String catalog = resultSet.getString("Catalog");
+            String schema = resultSet.getString("Schema");
+            String table = resultSet.getString("Table");
+            ClientTypeSignature clientTypeSignature = getClientTypeSignatureFromTypeString(resultSet.getString("Type"));
+            ColumnInfo.Builder builder = new ColumnInfo.Builder()
+                    .setColumnName(columnName)
+                    .setColumnLabel(columnName)
+                    .setCatalogName(catalog)
+                    .setSchemaName(schema)
+                    .setTableName(table)
+                    .setColumnTypeSignature(clientTypeSignature)
+                    .setNullable(ColumnInfo.Nullable.UNKNOWN);
+            setTypeInfo(builder, clientTypeSignature);
+            list.add(builder.build());
+        }
+        return list.build();
+    }
+
+    @VisibleForTesting
+    static ClientTypeSignature getClientTypeSignatureFromTypeString(String type)
+    {
+        Matcher topLevelMatcher = topLevelTypeRegexPattern.matcher(type);
+        String topLevelType;
+        List<ClientTypeSignatureParameter> arguments = new ArrayList<>();
+        if (topLevelMatcher.find()) {
+            topLevelType = topLevelMatcher.group(1);
+            String typeParameters = topLevelMatcher.group(2);
+            if (topLevelType.equals("decimal")) {
+                String[] precisionAndScale = typeParameters.split(",");
```

Comment:
We prefer to use Guava `Splitter` (which doesn't use a regex):
```java
List<String> precisionAndScale = Splitter.on(',').splitToList(typeParameters);
checkArgument(precisionAndScale.size() == 2, "Invalid decimal parameters: " + typeParameters);
arguments.add(ClientTypeSignatureParameter.ofLong(parseLong(precisionAndScale.get(0))));
arguments.add(ClientTypeSignatureParameter.ofLong(parseLong(precisionAndScale.get(1))));
```
`parseLong` is a good method to static import since the name is self descriptive (the `Long.` prefix doesn't add any value)